src/explore/AB_testing/by_date_camps.py

```diff
@@ -17,63 +17,80 @@
 4. Bandit (the models) probabilities are updated as they are played 
 '''
  
-
-from cap_ab_testing import experiment_numerical
-from cap_ab_testing_rand import experiment_numericalR
+from enum import Enum
+import logging
 import pandas as pd 
 import numpy as np 
 import matplotlib.pyplot as plt  
+import pickle, csv 
+
+from cap_ab_testing import experiment_numerical
+from cap_ab_testing_rand import experiment_numericalR
 from preprocessing_formatting import drop_cols , one_hot_encoding , scale
 from postHOCAB import run_tests 
-import pickle, csv 
+
 pd.set_option('display.max_columns', None) 
-dataframe = pd.read_csv('/home/allen/RIP_Tensor1/capstone2/Capstone2/src/explore/ab_df.csv') 
-#ids = pd.read_csv('/home/allen/RIP_Tensor1/capstone2/Capstone2/src/explore/testing/x1.csv')
-df_withdates = pd.read_csv('/home/allen/RIP_Tensor1/capstone2/Capstone2/data/placeholder/df_withdates.csv') 
-df_blank = pd.read_csv('/home/allen/RIP_Tensor1/capstone2/Capstone2/data/ready12_24_train.csv') 
 
-def edit_df():
+DATAFRAME_AB = '/home/allen/RIP_Tensor1/capstone2/Capstone2/src/explore/ab_df.csv'
+DATAFRAME_WITHDATES = '/home/allen/RIP_Tensor1/capstone2/Capstone2/data/placeholder/df_withdates.csv'
+DATAFRAME_BLANK = '/home/allen/RIP_Tensor1/capstone2/Capstone2/data/ready12_24_train.csv'
+
+_LOGGER = logging
+
+
+class CampColumns(Enum):
+    START_DATE2 = 'Camp_Start_Date2'
+    END_DATE2 = 'Camp_End_Date2'
+    CAMP_ID = 'Health_Camp_ID'
+
+
+def edit_df(df):
     '''
     5/20 
     This is being edited since new csv was made. 
     '''
     to_del =  [ 'Unnamed: 0', 'Unnamed: 0.1',
     '1352', '1704', '1729', '2517', '2662', '23384',  '2100', '1036', '1216', '1217',
     'City_Type2_x','Job Type_x','Category 2','Category 3','Category 1', 'online_score']
-    df = dataframe.drop(to_del,axis=1) 
+    df = df.drop(to_del,axis=1) 
+    _LOGGER.info("datframe edited containing columns {}".format(df.columns))
     print(df.columns) 
+
     return df
 
+
 def sep_by_date(df_encode):
     '''
     input = df
     actions = Create dict of camp_IDs, compare each start date to end_date
     create tupple of camp_ID and other Campsz that end before camp_ID starts (IDs that can be used for model training)
     '''
-    start = df_withdates['Camp_Start_Date2'].values
-    end = df_withdates['Camp_End_Date2'].values
-    camp_id = df_encode['Health_Camp_ID'].values 
-    unique={}
+    start_dates = df_withdates[CampColumns.START_DATE2.value].values
+    end_dates = df_withdates[CampColumns.END_DATE2.value].values
+    camp_ids = df_encode[CampColumns.CAMP_ID.value].values 
+    unique = {}
     d_time = {}
-    for item in list(zip(start,end, camp_id)):
-        if item not in unique:
-            unique[item[2]] =item
```

Comment:
I don't think `item` would ever be in unique this way.  item is a size 3 tuple but unique is just indexed on camp_id?